File: week2_algorithmic_warmup/1_fibonacci_fast.js
const fibonacci_number_fast = function(n) {
  n = parseInt(n)
  if(n === 0) return 0
  if(n === 1) return 1
  const store = [0,1];
  for(i=1; i<n; i++){
    store.push(store[i-1] + store[i])
  }
  return store[store.length - 1];
}

data = '';

process.stdin.on('readable', function() {
  var chunk;
  while (chunk = process.stdin.read()) {
    data += chunk;
  }
});

process.stdin.on('end', function () {

  let lines = data.split(/\r\n|\r|\n|\\n/)

  console.log(fibonacci_number_fast(lines[0]))

});

File: week2_algorithmic_warmup/1_fibonacci_fast.py
# Uses python3
import sys
import random

def fibonacci_number_fast(n):
    if n == 0:
        return 0
    if n == 1:
        return 1
    store = [0,1]
    for j in range(2, n+1):
        store.append(store[len(store)-2] + store[len(store)-1])
    return store.pop()

if __name__ == '__main__':

    input = sys.stdin.read()
    n = int(input)

    print(fibonacci_number_fast(n))
